src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -41,19 +41,13 @@
  * {@code FileReader}.
  *
  * @apiNote
- * To release resources used by this stream {@link #close} should be called
- * directly or by try-with-resources. Subclasses are responsible for the cleanup
- * of resources acquired by the subclass.
- * Subclasses that override {@link #finalize} in order to perform cleanup
- * should be modified to use alternative cleanup mechanisms such as
- * {@link java.lang.ref.Cleaner} and remove the overriding {@code finalize} method.
+ * The {@link #close} method should be called to release resources used by this
+ * stream, either directly, or by try-with-resources.
```

Comment:
The updated wording looks much better. It pre-dates your change but maybe "or by try-with-resources" should be changed to "with the {@code try}-with-resources statement" while you are there.

Reply:
Good idea